<launch>
  <arg name="input_raw_image" default="/image_raw" />
  <arg name="input_nearest_roi_rect" default="/feat_proj_bbox/nearest_roi_rect" />
  <!-- <arg name="input_yolo_detected_boxes" default="/detection/image_detector/objects" /> -->
  <arg name="input_yolo_detected_boxes" default="/transform_msgs_jsk2autoware/output" />

  <arg name="debug_log" default="false" />
  <arg name="debug_view" default="false" />
  <arg name="kernel_sigma" default="0.5" />
  <arg name="cell_size" default="4" />
  <arg name="num_scales" default="14" />
  <arg name="offset" default="10" />
  <arg name="interpolation_frequency" default="2" />

  <node name="kcf_tracker"
        pkg="nodelet" type="nodelet"
        output="screen"
        args="standalone kcf_ros/KcfTrackerROS">
    <remap from="~input_raw_image" to="$(arg input_raw_image)" />
    <remap from="~input_nearest_roi_rect" to="$(arg input_nearest_roi_rect)" />
    <remap from="~input_yolo_detected_boxes" to="$(arg input_yolo_detected_boxes)" />
    <rosparam subst_value="true">
      debug_log: $(arg debug_log)
      debug_view: $(arg debug_view)
      kernel_sigma: $(arg kernel_sigma)
      cell_size: $(arg cell_size)
      num_scales: $(arg num_scales)
      offset: $(arg offset)
      interpolation_frequency: $(arg interpolation_frequency)
      approximate_sync: true
    </rosparam>
  </node>

</launch>
